### PR TITLE
Run Keep Alive Framework on strict.

### PR DESCRIPTION
**Pull request by jeschkies**
Summary:
Configures and packages the keep-alive-framework to run on struct
and thus on soak cluster.

Relates to mesosphere/soak-cluster-configurations#577.

JIRA issues: DCOS-47868

### CODE REVIEW

File: examples/keep-alive-framework/src/main/scala/com/mesosphere/usi/helloworld/KeepAlivePodSpecHelper.scala [hunk 1]
```diff
@@ -13,7 +13,7 @@ object KeepAlivePodSpecHelper {
   val runSpec: RunTemplate = RunTemplate(
     resourceRequirements = List(ScalarRequirement(ResourceType.CPUS, 0.001), ScalarRequirement(ResourceType.MEM, 32)),
     shellCommand = """echo "Hello, world" && sleep 30""",
-    role = "test"
+    role = "slave_public"
```
Comment (zen-dog) [inline examples/keep-alive-framework/src/main/scala/com/mesosphere/usi/helloworld/KeepAlivePodSpecHelper.scala]: wait, why?

File: examples/keep-alive-framework/src/main/scala/com/mesosphere/usi/helloworld/KeepAliveMesosClientFactory.scala [hunk 2]
```diff
@@ -3,32 +3,29 @@ package com.mesosphere.usi.helloworld
...
   val frameworkInfo = FrameworkInfo
     .newBuilder()
-    .setUser(
-      new SystemProperties()
-        .get("user.name")
-        .getOrElse(throw new IllegalArgumentException("A local user is needed to launch Mesos tasks")))
+    .setUser("root")
     .setName("KeepAliveFramework")
-    .addRoles("test")
+    .addRoles("slave_public")
```
Comment (zen-dog) [inline examples/keep-alive-framework/src/main/scala/com/mesosphere/usi/helloworld/KeepAliveMesosClientFactory.scala]: why do you need to use `slave_public`?

Comment (jeschkies) [reply examples/keep-alive-framework/src/main/scala/com/mesosphere/usi/helloworld/KeepAliveMesosClientFactory.scala]: That's a left over from my tests. Offer accepts would not get through.

File: examples/keep-alive-framework/build.gradle [hunk 3]
```diff
@@ -14,3 +16,14 @@ dependencies {
...
+task uploadTarball(type: S3Upload) {
+    description = "Upload the framework tarball to S3."
+    group = "CI"
+    dependsOn distTar
+
+    region = "us-west-2"
+    bucket = "usi-builds"
+    prefix = "mesosphere/usi/artifacts"
+    source = files(distTar.archiveFile).asFileTree
+}
```
Comment (jeschkies) [inline examples/keep-alive-framework/build.gradle:29]: We might want to investigate [Gradle Publishing](https://docs.gradle.org/current/userguide/distribution_plugin.html#sec:publishing_distributions) but this should suffice to get the latest master build on Soak.

File: examples/keep-alive-framework/src/main/scala/com/mesosphere/usi/helloworld/KeepAlivePodSpecHelper.scala (hunk 1 above)
Comment (timcharper) [reply examples/keep-alive-framework/src/main/scala/com/mesosphere/usi/helloworld/KeepAlivePodSpecHelper.scala]: Agreed, we should be able use whichever role we like.

File: examples/keep-alive-framework/keep-alive-framework-app.json [hunk 4]
```diff
@@ -0,0 +1,60 @@
+{
+  "env": {
+  },
+  "id": "/usi/keep-alive-framework",
+  "backoffFactor": 1.15,
+  "backoffSeconds": 1,
+  "cmd": "export JAVA_HOME=$(ls -d $MESOS_SANDBOX/jdk*/jre/); export JAVA_HOME=${JAVA_HOME%/}; export PATH=$(ls -d $JAVA_HOME/bin):$PATH && export JAVA_OPTS=\"-Xms256M -Xmx512M\"; ./keep-alive-framework-0.1/bin/keep-alive-framework https://master.mesos https://leader.mesos:5050 $MESOS_SANDBOX/.ssl/ca-bundle.crt $MESOS_SANDBOX/private_key.pem strict-usi",
```
Comment (jeschkies) [inline examples/keep-alive-framework/keep-alive-framework-app.json]: DC/OS and Mesos URLs are from https://github.com/dcos/cosmos/blob/master/cosmos-server/src/main/scala/com/mesosphere/cosmos/package.scala#L88-L98.

File: examples/keep-alive-framework/build.gradle (hunk 3 above)
Comment (jeschkies) [reply examples/keep-alive-framework/build.gradle:29]: I'd love to trigger a restart on soak but it does not seem trivial to authenticate. Checkout https://github.com/mesosphere/soak-cluster-configurations/pull/506 for the app.

File: examples/keep-alive-framework/keep-alive-framework-app.json [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+  "disk": 0,
+  "fetch": [
+    {
+      "uri": "https://s3-us-west-2.amazonaws.com/usi-builds/mesosphere/usi/artifacts/keep-alive-framework-0.1.tar",
+      "extract": true,
+      "executable": false,
+      "cache": false
+    },
+    {
+      "uri": "https://downloads.mesosphere.com/java/server-jre-8u192-linux-x64.tar.gz",
```
Comment (timcharper) [inline examples/keep-alive-framework/keep-alive-framework-app.json]: we're moving away from the Oracle jdk; maybe we should just have this run on openjdk?

**Review by timcharper — APPROVED** [review]

File: examples/keep-alive-framework/keep-alive-framework-app.json (hunk 5 above)
Comment (jeschkies) [reply examples/keep-alive-framework/keep-alive-framework-app.json]: Good point. This should be Amazon Corretto to be save.

**Comment by jeschkies** [comment]
Ping @zen-dog

**Review by zen-dog — APPROVED** [review]

**Pull request merged**